lib/protocol/websocket/connection.rb

```diff
@@ -191,7 +193,16 @@ def read
 				
 				while read_frame
 					if @frames.last&.finished?
-						buffer = @frames.map(&:unpack).join
+						buffer = @frames.map(&:unpack).join("")
+						
+						if @frames.first.is_a?(TextFrame)
```

Comment:
I would suggest something more like:

```
buffer = @frames.first.decode(buffer)
```

Reply:
There is something I do not understand here, is there a `Frame#decode` method?

Reply:
Nope, I'm suggesting we introduce it. The frame class should know how to process the data.